pkg/osm/control_plane.go

```diff
@@ -0,0 +1,53 @@
+package osm
+
+import (
+	"context"
+
+	"github.com/openservicemesh/osm/pkg/catalog"
+	"github.com/openservicemesh/osm/pkg/certificate"
+	"github.com/openservicemesh/osm/pkg/envoy"
+	"github.com/openservicemesh/osm/pkg/envoy/registry"
+	"github.com/openservicemesh/osm/pkg/messaging"
+	"github.com/openservicemesh/osm/pkg/workerpool"
+)
+
+const (
+	workerPoolSize = 0
+)
+
+type ProxyConfigServer[T any] interface {
```

Comment:
Could you add some comments (in the code or PR description) about what these types represent and why they're generic

Reply:
Yup, good call!